contracts/AdjudicationFramework.sol

```diff
@@ -91,34 +101,44 @@ contract AdjudicationFramework is BalanceHolder {
 
     mapping(bytes32 => ArbitrationRequest) public question_arbitrations;
 
-    // TODO: Work out how this is implemented in xdai or whatever we use
-    modifier l1_forkmanager_only() {
-        require(msg.sender == address(bridge), "Message must come from bridge");
-        require(
-            bridge.messageSender() == FORK_MANAGER_SPECIAL_ADDRESS,
-            "Message must come from L1 ForkManager"
-        );
-        _;
-    }
-
+    /// @param _realityETH The reality.eth instance we adjudicate for
+    /// @param _dispute_fee The dispute fee we charge reality.eth users
+    /// @param _forkArbitrator The arbitrator contract that escalates to an L1 fork, used for our governance
+    /// @param _initialArbitrators Arbitrator contracts we initially support
     constructor(
         address _realityETH,
         uint256 _dispute_fee,
-        IAMB _bridge,
-        address[] memory _initial_arbitrators
+        address _forkArbitrator,
+        address[] memory _initialArbitrators
     ) {
-        realityETH = IRealityETH_ERC20(_realityETH);
+        realityETH = IRealityETH(_realityETH);
         dispute_fee = _dispute_fee;
-        bridge = _bridge;
-        for (uint256 i = 0; i < _initial_arbitrators.length; i++) {
-            arbitrators[_initial_arbitrators[i]] = true;
+        forkArbitrator = _forkArbitrator;
+
+        // Create reality.eth templates for our add and remove questions
+        // We'll identify ourselves in the template so we only need a single parameter for questions, the arbitrator in question.
+        // TODO: We may want to specify a document with the terms that guide this decision here, rather than just leaving it implicit.
+
+        string memory templatePrefixAdd = '{"title": "Should we add arbitrator %s to the framework ';
+        string memory templatePrefixRemove = '{"title": "Should we remove arbitrator %s from the framework ';
+        string memory templateSuffix = '?", "type": "bool", "category": "adjudication", "lang": "en"}';
+
+        string memory addTemplate = _toString(abi.encodePacked(templatePrefixAdd, address(this), templateSuffix));
+        string memory removeTemplate = _toString(abi.encodePacked(templatePrefixRemove, address(this), templateSuffix));
+
+        templateIdAddArbitrator = realityETH.createTemplate(addTemplate);
+        templateIdRemoveArbitrator = realityETH.createTemplate(removeTemplate);
+
+        for (uint256 i = 0; i < _initialArbitrators.length; i++) {
+           arbitrators[_initialArbitrators[i]] = true;
         }
+
     }
 
     /// @notice Return the dispute fee for the specified question. 0 indicates that we won't arbitrate it.
     /// @dev Uses a general default, but can be over-ridden on a question-by-question basis.
     function getDisputeFee(bytes32) public view returns (uint256) {
-        // Todo: make it depend on the question
+        // TODO: Should we have a governance process to change this?
```

Comment:
Yeah, there is also the overall question how to govern the infrastructure contracts deployed on L2. 

I think its okay, if during a fork we just change the state of L2 (i.e. code that describes the contracts) just on L1 via the normal forking update. But I have not though enough about it

Reply:
I think if they need governing we can just do it the same way we add and remove arbitrators. The question is just whether it should be governed, or whether it's better to fix it based on some multiplier of the forking fee, or whether it should be governed but within some predetermined parameters. Probably just do the simplest thing for now, we can always deploy new arbitration frameworks later.